src/components/tx-flow/flows/NftTransfer/index.tsx

```diff
@@ -4,31 +4,32 @@ import useTxStepper from '../../useTxStepper'
 import SendNftBatch from './SendNftBatch'
 import ReviewNftBatch from './ReviewNftBatch'
 
-export type NftTransferParams = {
+type NftTransferFlowProps = {
   txNonce?: number
   recipient?: string
   tokens?: SafeCollectibleResponse[]
 }
 
-export type SubmittedNftTransferParams = NftTransferParams & {
+export type NftTransferParams = {
   recipient: string
   tokens: SafeCollectibleResponse[]
 }
 
-type NftTransferFlowProps = NftTransferParams & {
-  txNonce?: number
+const defaultParams: NftTransferParams = {
+  recipient: '',
+  tokens: [],
 }
 
 const NftTransferFlow = ({ txNonce, ...params }: NftTransferFlowProps) => {
-  const { data, step, nextStep, prevStep } = useTxStepper<[NftTransferParams, SubmittedNftTransferParams | undefined]>([
-    params,
-    undefined,
-  ])
+  const { data, step, nextStep, prevStep } = useTxStepper<NftTransferParams>({
+    ...defaultParams,
+    ...params,
+  })
 
   const steps = [
-    <SendNftBatch key={0} params={data[0]} onSubmit={(formData) => nextStep([formData, formData])} />,
+    <SendNftBatch key={0} params={data} onSubmit={(formData) => nextStep({ ...data, ...formData })} />,
```

Comment:
If we no longer want to separate the data of each step, I would suggest we instead merge inside the hook.

I'd be cautious of this approach though as the types are less strict per step and we run the risk of overwriting same-named fields.

Reply:
A separate improvement would be passing the type as a generic to the component and then using that inside, e.g.

```ts
const SendNftBatch = <T extends Object>() => {
  const formMethods = useForm<T>({ ... })

  return (
    ...
  )
}

<SendNftBatch<NftTransferParams> {...props} />
```

Reply:
The stepper hook should just manage steps and hold whatever data is passed from outside. It shouldn't have heuristics about how the data flows between steps.